src/apps/testapps/testPolyfill.c

```diff
@@ -41,14 +42,68 @@ static GeoCoord emptyVerts[] = {{0.659966917655, -2.1364398519394},
 static Geofence emptyGeofence = {.numVerts = 3, .verts = emptyVerts};
 static GeoPolygon emptyGeoPolygon;
 
-static int countActualHexagons(H3Index* hexagons, int numHexagons) {
-    int actualNumHexagons = 0;
-    for (int i = 0; i < numHexagons; i++) {
-        if (hexagons[i] != 0) {
-            actualNumHexagons++;
+/**
+ * Return true if the cell crosses the meridian.
+ */
+static bool isTransmeridianCell(H3Index h) {
+    GeoBoundary bndry;
+    H3_EXPORT(h3ToGeoBoundary)(h, &bndry);
+
+    double minLon = M_PI, maxLon = -M_PI;
+    for (int i = 0; i < bndry.numVerts; i++) {
+        if (bndry.verts[i].lon < minLon) minLon = bndry.verts[i].lon;
+        if (bndry.verts[i].lon > maxLon) maxLon = bndry.verts[i].lon;
+    }
+
+    return maxLon - minLon > M_PI - (M_PI / 4);
+}
+
+static void fillIndex_assertions(H3Index h) {
+    if (isTransmeridianCell(h)) {
+        // TODO: these do not work correctly
```

Comment:
Is this true for all transmeridian cells, or just polar cells? I thought transmeridian cells in general should work correctly.

Reply:
I recall this was an issue for not only the cells at the poles.

Reply:
If so, that's a problem. The `polyfill` algo is supposed to work correctly for transmeridian polygons, and we should have tests to confirm.

Reply:
I tried replacing `isTransmeridianCell` with this body, and the test suite failed:
```
    int res = H3_GET_RESOLUTION(h);
    GeoCoord northPoleCoords = {.lat = M_PI_2, .lon = 0};
    H3Index northPoleIndex = H3_EXPORT(geoToH3)(&northPoleCoords, res);
    if (northPoleIndex == h) {
        return true;
    }
    GeoCoord southPoleCoords = {.lat = -M_PI_2, .lon = 0};
    H3Index southPoleIndex = H3_EXPORT(geoToH3)(&southPoleCoords, res);
    if (southPoleIndex == h) {
        return true;
    }
    return false;
```